ecosystem/sep-0024.md

```diff
@@ -170,26 +171,183 @@ There are several possible kinds of response, depending on whether the anchor ne
+
+##### Wallet Feature: Claimable Balance Polling
+
+Polling allows the wallet to detect claimable balances and provide a UI to users that there is a balance available to claim. See the ["Claim Account Claimable Balance"](#wallet-feature-claim-account-claimable-balance) section to learn more about how to claim a balance.
+There are a few options as to how a wallet can poll and surface claimable balance information for their users.
+1. The wallet can poll the anchor's [transaction history endpoints](#anchor-feature-extend-transaction-history-endpoints-for-claimable-balances)
+
+2. The wallet can poll Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/list/) endpoint (or Horizon's [`/claimable_balances/`](https://developers.stellar.org/api/resources/claimablebalances/single/) single endpoint) to pull any available claimable balances for the claimant.
+
+   * The single endpoint needs the `claimable_balance_id` in order to retrieve the claimable balance info.
+     * This can be obtained by extracting the `claimable_balance_id` from `results_xdr` of the deposit transaction submitted if the anchor does not extend their [transaction history endpoints](#anchor-feature-extend-transaction-history-endpoints-for-claimable-balances)
+
+##### Wallet Feature: Claim Account Claimable Balance
+
+It's recommended that Wallets provides a UI to enable accounts to claim their balances [polled](#wallet-feature-claimable-balance-polling)
+
+[Example on how to claim a balance](https://developers.stellar.org/docs/glossary/claimable-balance/#claim-claimable-balance)
+Accounts will still need to establish trustline to the asset before claiming the outstanding balance.
+
+##### Anchor Claimable Balance Flow
+
+1. Wallets make a request to `/transactions/deposit/interactive` and provide the `claimable_balance_supported` request parameter.
+2. Anchors mark the transaction as [`claimable_balance_supported=true`](#deposit)
+3. Anchors detect that the user's Stellar account doesn't have a trustline.
+4. The anchor should then report transaction status as `pending_anchor` and submit a Stellar transaction containing a create claimable balance operation
+5. From there the anchor will continue the [deposit flow](#basic-anchor-implementation)
+
+##### Anchor Feature: Extend Transaction history endpoints For Claimable Balances
+
+Wallets query the [transaction endpoint](#transaction-history) to see and display the status of deposits.
+Anchors should extend their transaction(s) endpoint response to include the `claimable_balance_id` that the anchor added to the [transaction field](####Fields-for-deposit-transactions).
+This will help wallets to provide UI to showcase that their user's deposit was successful and theres an [available claimable balance for them.](#wallet-feature-claimable-balance-polling)
+
+```
+GET TRANSFER_SERVER_SEP0024/transactions
+```
```

Comment:
I don't think these examples are necessary, we already give json examples in the transaction section. Just add `null` and the non-null value to those examples. Anchors who don't implement claimable balances will either not add the attribute or keep it `null`.